time/src/main/java/net/jqwik/time/internal/properties/arbitraries/DefaultLocalDateArbitrary.java

```diff
@@ -89,11 +112,19 @@ private int latestAllowedMonth() {
 	private LocalDate effectiveMinDate() {
 		LocalDate effective = dateBetween.getMin() == null ? DEFAULT_MIN_DATE : dateBetween.getMin();
 		int earliestMonth = earliestAllowedMonth();
+		int latestMonth = latestAllowedMonth();
+		if (latestMonth < effective.getMonth().getValue()) {
+			effective = effective.withDayOfMonth(1).withMonth(1).plusYears(1);
+		}
 		if (earliestMonth > effective.getMonth().getValue()) {
-			return effective.withMonth(earliestMonth);
+			effective = effective.withDayOfMonth(1).withMonth(earliestMonth);
 		}
 		if (dayOfMonthBetween.getMin() != null && dayOfMonthBetween.getMin() > effective.getDayOfMonth()) {
-			return effective.withDayOfMonth(dayOfMonthBetween.getMin());
+			try {
+				effective = effective.withDayOfMonth(dayOfMonthBetween.getMin());
+			} catch (DateTimeException e) {
```

Comment:
Try-catch as case differentiation is a smell. Can you instead test for the condition that will lead to the exception?
If that is too much work, at least document why the exception can occur here.

Reply:
Done